### ISSUE TITLE
Import GKE, EKS External Clusters 

### ISSUE THREAD

**Issue by ahmadhamzh** [kubermatic/dashboard#6332]
### What happened
when try to Import externals Cluster (that not been created by our dashboard) we get this error .

* EKS
`Error: Can't access cluster via kubeconfig,error: the server has asked for the client to provide credentials`.

![image](https://github.com/kubermatic/dashboard/assets/85109141/91137e24-8e0a-4095-9ce6-5b5b9eb3d39e)

* GKE
 'zone' field cannot be used to access GKE regional clusters. Use 'name' field instead` .

![image](https://github.com/kubermatic/dashboard/assets/85109141/2751328e-d300-4196-b632-e9dbaa6128f2)

### Expected behavior
import external cluster should work even if the cluster been created from the provider side.
### How to reproduce
try to create a GKE or EKS cluster from the provider side and then try to import it

**Comment by xrstf** [comment]
The GKE issue only happens for _regional_ clusters. Zonal clusters work fine.

**Comment by xrstf** [comment]
Regarding GKE: The Go API we're using (https://pkg.go.dev/google.golang.org/api@v0.148.0/container/v1) seems to simply not support regional clusters. I tried manually changing the zone parameter to be a zone and not a region (i.e. my cluster lives in the eu-north1 region, and I modified the request to use eu-north1-a, but go another error).

The Go module says

> When possible, we recommend using our newer [Cloud Client Libraries for Go](https://pkg.go.dev/cloud.google.com/go) that are still actively being worked and iterated on.

and https://pkg.go.dev/cloud.google.com/go/container@v1.26.1/apiv1 does have https://pkg.go.dev/cloud.google.com/go/container@v1.26.1/apiv1/containerpb#GetClusterRequest, which could maybe work with regional clusters.

Not sure if and when we want to make the investment to rewrite our GKE integration to this code API.
